beacon_node/beacon_chain/src/sync_committee_rewards.rs

```diff
@@ -0,0 +1,79 @@
+use crate::{BeaconChain, BeaconChainError, BeaconChainTypes};
+
+use eth2::lighthouse::SyncCommitteeReward;
+use state_processing::per_block_processing::altair::sync_committee::compute_sync_aggregate_rewards;
+use types::{BeaconBlockRef, BeaconState, ExecPayload};
+use std::collections::HashMap;
+
+impl<T: BeaconChainTypes> BeaconChain<T> {
+    pub fn compute_sync_committee_rewards<Payload: ExecPayload<T::EthSpec>>(
+        &self,
+        block: BeaconBlockRef<'_, T::EthSpec, Payload>,
+        state: &mut BeaconState<T::EthSpec>,
+    ) -> Result<Vec<SyncCommitteeReward>, BeaconChainError> {
+        if block.slot() != state.slot() {
+            return Err(BeaconChainError::BlockRewardSlotError);
+        }
+
+        let spec = &self.spec;
+
+        let sync_aggregate = block
+            .body()
+            .sync_aggregate()?;
+
+        let sync_committee = state
+            .current_sync_committee()?
+            .clone();
+
+        let sync_committee_indices = state
+            .get_sync_committee_indices(&sync_committee)?;
+
+        let (participant_reward_value, proposer_reward_per_bit) = compute_sync_aggregate_rewards(&state, spec)
+            .map_err(|_| BeaconChainError::SyncCommitteeRewardsSyncError)?;
```

Comment:
Lets log this error for debugging purposes (this is how I found the cache issue above). Something like:

```rust
let (participant_reward_value, proposer_reward_per_bit) =
            compute_sync_aggregate_rewards(&state, spec).map_err(|e| {
                error!(
                    self.log,
                    "Error calculating sync rewards";
                    "error" => ?e,
                );
                BeaconChainError::SyncCommitteeRewardsSyncError
            })?;
```

Alternatively we can return in the error in the `BeaconChainError` (add the error type of `compute_sync_aggregate_rewards` as an inner field).